clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3178,6 +3178,13 @@ static void RenderFloatingPointOptions(const ToolChain &TC, const Driver &D,
       StringRef Val = A->getValue();
       if (Val == "fast" || Val == "on" || Val == "off" ||
           Val == "fast-honor-pragmas") {
+        // fast-honor-pragmas is deprecated -- replace it with fast
+        if (Val == "fast-honor-pragmas") {
+          D.Diag(diag::warn_drv_deprecated_arg)
+            << A->getAsString(Args) << /*hasReplacement=*/true
+            << "-ffp-contract=fast";
+          Val = "fast";
+        }
```

Comment:
I'm not sure that actually diagnosing this is a good idea.  We have to support it forever; let's just silently consider an alias.

Reply:
Can we document it as deprecated (or just remove it from the documentation) but silently accept it? I was copying this code from another option where an option was deprecated. I certainly don't see a problem with leaving it permanently.